engine/src/main/java/org/kigalisim/engine/support/RecalcKit.java

```diff
@@ -0,0 +1,66 @@
+/**
+ * Container for dependencies needed by recalculation strategies.
+ *
+ * <p>This class provides all the dependencies that recalculation strategies need
+ * to perform their operations, eliminating the need to cast Engine to SingleThreadEngine.</p>
+ *
+ * @license BSD-3-Clause
+ */
+
+package org.kigalisim.engine.support;
+
+import java.util.Optional;
+import org.kigalisim.engine.number.UnitConverter;
+import org.kigalisim.engine.state.ConverterStateGetter;
+import org.kigalisim.engine.state.StreamKeeper;
+
+/**
+ * Container for dependencies needed by recalculation strategies.
+ */
+public class RecalcKit {
+
+  private final Optional<StreamKeeper> streamKeeper;
```

Comment:
Let's change these all to required to simplify logic in the strategies so they do not have to check if empty or null. Please update all RecalcStrategy implementors accordingly.